swarm/storage/mru/resource.go

```diff
@@ -889,64 +827,58 @@ func (self *Handler) update(ctx context.Context, name string, data []byte, multi
 	return key, nil
 }
 
-// Closes the datastore.
+// Close closes the datastore.
 // Always call this at shutdown to avoid data corruption.
-func (self *Handler) Close() {
-	self.chunkStore.Close()
+func (h *Handler) Close() {
+	h.chunkStore.Close()
 }
 
 // gets the current block height
-func (self *Handler) getBlock(ctx context.Context, name string) (uint64, error) {
-	blockheader, err := self.headerGetter.HeaderByNumber(ctx, name, nil)
+func (h *Handler) getBlock(ctx context.Context, name string) (uint64, error) {
+	blockheader, err := h.headerGetter.HeaderByNumber(ctx, name, nil)
 	if err != nil {
 		return 0, err
 	}
 	return blockheader.Number.Uint64(), nil
 }
 
-// Calculate the period index (aka major version number) from a given block number
-func (self *Handler) BlockToPeriod(name string, blocknumber uint64) (uint32, error) {
-	return getNextPeriod(self.resources[name].startBlock, blocknumber, self.resources[name].frequency)
+// BlockToPeriod calculates the period index (aka major version number) from a given block number
+func (h *Handler) BlockToPeriod(name string, blocknumber uint64) (uint32, error) {
+	return getNextPeriod(h.resources[name].startBlock, blocknumber, h.resources[name].frequency)
 }
 
-// Calculate the block number from a given period index (aka major version number)
-func (self *Handler) PeriodToBlock(name string, period uint32) uint64 {
-	return self.resources[name].startBlock + (uint64(period) * self.resources[name].frequency)
+// PeriodToBlock calculates the block number from a given period index (aka major version number)
+func (h *Handler) PeriodToBlock(name string, period uint32) uint64 {
+	return h.resources[name].startBlock + (uint64(period) * h.resources[name].frequency)
 }
 
 // Retrieves the resource index value for the given nameHash
-func (self *Handler) get(nameHash string) *resource {
-	self.resourceLock.RLock()
-	defer self.resourceLock.RUnlock()
-	rsrc := self.resources[nameHash]
+func (h *Handler) get(nameHash string) *resource {
+	h.resourceLock.RLock()
+	defer h.resourceLock.RUnlock()
+	rsrc := h.resources[nameHash]
 	return rsrc
 }
 
 // Sets the resource index value for the given nameHash
-func (self *Handler) set(nameHash string, rsrc *resource) {
-	self.resourceLock.Lock()
-	defer self.resourceLock.Unlock()
-	self.resources[nameHash] = rsrc
+func (h *Handler) set(nameHash string, rsrc *resource) {
+	h.resourceLock.Lock()
+	defer h.resourceLock.Unlock()
+	h.resources[nameHash] = rsrc
 }
 
 // used for chunk keys
-func (self *Handler) resourceHash(period uint32, version uint32, namehash common.Hash) storage.Address {
-	// format is: hash(period|version|namehash)
-	hasher := self.hashPool.Get().(storage.SwarmHash)
-	defer self.hashPool.Put(hasher)
+func (h *Handler) resourceHash(period uint32, version uint32, nameHash common.Hash, publicKeyBytes [publicKeyLength]byte) storage.Address {
```

Comment:
what does this do? i dont get why we hashing a hash ?

Reply:
It creates the update key .. um .. `address`